File: src/main/java/com/dmk/chooser/Chooser.kt
package com.dmk.chooser

import java.io.File
import java.util.*

class Chooser(libPath: String) {

    private val albumMap = LinkedHashMap<String, String>()

    init {
        map(File(libPath))
    }

    private fun map(inFile: File) {
        if (musicFile(inFile.name)) {
            albumMap[inFile.parentFile.name] = inFile.parentFile.absolutePath
        } else {
            val file = inFile.listFiles()
            if (file != null) {
                for (aFile in file) {
                    if (musicFile(aFile.name)) {
                        map(aFile)
                        break
                    } else {
                        map(aFile)
                    }
                }
            }
        }
    }

    private fun musicFile(fileName: String): Boolean {
        return fileName.endsWith("mp3") or
                fileName.endsWith("flac") or
                fileName.endsWith("mp4") or
                fileName.endsWith("m4p") or
                fileName.endsWith("wav") or
                fileName.endsWith("alac") or
                fileName.endsWith("wma") or
                fileName.endsWith("aac") or
                fileName.endsWith("m4a")
    }

    fun getRandomAlbumPath(): String? {
        val album = ArrayList(albumMap.keys)[Random().nextInt(albumMap.keys.size)]
        return albumMap[album]
    }
}

File: src/main/resources/chooser.py
import os
import random
import subprocess

dir_path = os.path.dirname(os.path.realpath(__file__))

propertiesPath = dir_path + "\\settings.properties"
musicLibPath: str
musicPlayerPath: str

with open(propertiesPath) as f:
    content = f.readlines()
    content = [x.strip() for x in content]
    for line in content:
        splitted = line.split("=")
        propName = splitted[0]
        propValue = splitted[1]
        if propName == 'chooser.musicLibraryPath':
            musicLibPath = propValue
        elif propName == 'chooser.playerPath':
            musicPlayerPath = propValue

print("music library :", musicLibPath)
print("music player path :", musicPlayerPath)

albumsList = []

for subdir, dirs, files in os.walk(musicLibPath):
    for file in files:
        if file.endswith(("mp3",
                          "flac",
                          "mp4",
                          "m4p",
                          "wav",
                          "alac",
                          "wma",
                          "aac",
                          "m4a")):
            albumsList.append(subdir)
            break
for x in albumsList:
    print(x)
size = len(albumsList)
print("albums found :", size)
selectedAlbumIndex = random.randint(0, size)
print("selected album index :", selectedAlbumIndex)
selectedAlbumPath = albumsList[selectedAlbumIndex]
print("selected album :", selectedAlbumPath)

args = [musicPlayerPath, selectedAlbumPath]
subprocess.call(args)
